src/main/kotlin/it/pagopa/ecommerce/scheduler/services/NodeService.kt

```diff
@@ -38,33 +41,70 @@ class NodeService(
         .awaitSingleOrNull()
         ?: throw TransactionEventNotFoundException(transactionId, transactionActivatedEventCode)
 
-    val transactionAuthRequestedEventCode =
-      TransactionEventCode.TRANSACTION_AUTHORIZATION_REQUESTED_EVENT
+    // Retrieved to check if the user canceled event request exists
+    val userCanceledEvent =
+      transactionsEventStoreRepository
+        .findByTransactionIdAndEventCode(
+          transactionId.toString(), TransactionEventCode.TRANSACTION_USER_CANCELED_EVENT)
+        .cast(TransactionUserCanceledEvent::class.java)
+        .awaitSingleOrNull()
 
     val authEvent =
       transactionsEventStoreRepository
         .findByTransactionIdAndEventCode(
-          transactionId.toString(), transactionAuthRequestedEventCode)
+          transactionId.toString(), TransactionEventCode.TRANSACTION_AUTHORIZATION_REQUESTED_EVENT)
         .cast(TransactionAuthorizationRequestedEvent::class.java)
         .awaitSingleOrNull()
-        ?: throw TransactionEventNotFoundException(transactionId, transactionAuthRequestedEventCode)
-
-    logger.info("Invoking closePayment with outcome {}", transactionOutcome)
 
     val closePaymentRequest =
-      ClosePaymentRequestV2OKDto().apply {
-        paymentTokens = activatedEvent.data.paymentNotices.map { it.paymentToken }
-        outcome = transactionOutcome
-        idPSP = authEvent.data.pspId
-        paymentMethod = authEvent.data.paymentTypeCode
-        idBrokerPSP = authEvent.data.brokerName
-        idChannel = authEvent.data.pspChannelCode
-        this.transactionId = transactionId.toString()
-        totalAmount = (authEvent.data.amount + authEvent.data.fee).toBigDecimal()
-        fee = authEvent.data.fee.toBigDecimal()
-        timestampOperation = OffsetDateTime.now()
-        additionalPaymentInformations = mapOf()
-      }
-    return nodeClient.closePayment(closePaymentRequest).awaitSingle()
+      mono { userCanceledEvent }
```

Comment:
I'd make it more explicit that this method only handles closepayment KO (update the method name?), or do we generalize it.
Also I would remove the reference to the user cancellation event (in the future we could do a closepayment KO even without this event).
In summary, I would remove event references from this service.

Reply:
This method doesn't handle only closepaymentKO.
This is an evolution of the existing method, used in retry close payment consumer.
What I added is a logic to check if the transaction has the right status before proceed to the closePayment.
I could extract this logic in another module, if you prefer to have a clean invocation to che close payment.